packages/nimble-components/src/wafer-map/modules/rendering.ts

```diff
@@ -11,17 +12,21 @@ export class RenderingModule {
     public constructor(private readonly wafermap: WaferMap) {}
 
     public updateSortedDiesAndDrawWafer(): void {
-        this.dies = this.wafermap.dataManager.diesRenderInfo.sort((a, b) => {
-            if (a.fillStyle > b.fillStyle) {
-                return 1;
-            }
-            if (b.fillStyle > a.fillStyle) {
-                return -1;
-            }
+        if (this.wafermap.dataManager instanceof DataManager) {
+            this.dies = this.wafermap.dataManager.diesRenderInfo.sort(
+                (a, b) => {
+                    if (a.fillStyle > b.fillStyle) {

```

Comment:
This was just meant to cluster all the values with the same fill style together, so we do not have to change the canvas context so often. this became less of a limitation over time and even a bottleneck and is one of the reasons we are doing a new rendering startegy

Reply:
I guess I still don't have a clear picture of whether this implementation is supposed to change. Do you intend to modify this so that you are making a numerical comparison, or is there going to no longer be a need to do this sort, or neither of those?

Reply:
@munteannatan, just making sure you are aware I unresolved this comment thread.

Reply:
Sorry, I did not get the notification in time.

In the new implementation we are firstly limiting the actual number of colors we display, like you see in this PR. This means a reduced number of fillstyles to change between.

Secondly, we are no longer storing the value for each die, we are working with a dictionary and keys instead, so we remove the data duplication as much as we can.

Thirdly, we are iterating over typed arrays instead of the list of dies, which is a lot faster and gives us improved rendering performance.

And finally, we are filtering dies out from rendering based on the zoom level. This is particularly important for big wafers, so we are implementing a caching technique where we are clustering based on the columns, so we can remove the whole column at once without iterating over the dies in the column at all, so the color clustering is no longer beneficial.